Display messages aren't still aligned

```go
package main

import (
	"testing"

	"github.com/paveg/diagassert"
)

func Test_Add(t *testing.T) {
	result := Add(2, 1)
	expected := 0

	diagassert.Assert(t, result == expected, diagassert.Values{"result": result, "expected": expected})
}

func Test_Minus(t *testing.T) {
	result := Minus(2, 1)
	expected := 0

	diagassert.Assert(t, result == expected, diagassert.Values{"result": result, "expected": expected})
}
```


```bash
❯ go test -v
=== RUN   Test_Add
    main_test.go:13: ASSERTION FAILED at main_test.go:13

          assert(result == expected)
                 |      |  |
                 30m         0

                        |
                        false

        CAPTURED VALUES:
          result = 3 (int)
          expected = 0 (int)

        [MACHINE_READABLE_START]
        EXPR: result == expected
        RESULT: false
        VARIABLES: expected=0,result=3
        EVALUATION_STEPS:
          Step 1: `result` => 3
          Step 2: `expected` => 0
          Step 3: `result == expected` with 3 == 0 => false
        CAPTURED_VALUES_START
        VALUE: result = 3 (int)
        VALUE: expected = 0 (int)
        CAPTURED_VALUES_END
        [MACHINE_READABLE_END]

--- FAIL: Test_Add (0.00s)
=== RUN   Test_Minus
    main_test.go:20: ASSERTION FAILED at main_test.go:20

          assert(result == expected)
                 |      |  |
                 10m         0

                        |
                        false

        CAPTURED VALUES:
          result = 1 (int)
          expected = 0 (int)

        [MACHINE_READABLE_START]
        EXPR: result == expected
        RESULT: false
        VARIABLES: expected=0,result=1
        EVALUATION_STEPS:
          Step 1: `result` => 1
          Step 2: `expected` => 0
          Step 3: `result == expected` with 1 == 0 => false
        CAPTURED_VALUES_START
        VALUE: result = 1 (int)
        VALUE: expected = 0 (int)
        CAPTURED_VALUES_END
        [MACHINE_READABLE_END]

--- FAIL: Test_Minus (0.00s)
FAIL
exit status 1
FAIL    example 0.263s
```

--- a/internal/formatter/visual.go
+++ b/internal/formatter/visual.go
@@ -340,50 +340,72 @@ func (f *VisualFormatter) colorizePerValuePipeLine(line string, layerAssignment 
 
 // buildColoredValueLine builds a value line with appropriate colors for each value
 func (f *VisualFormatter) buildColoredValueLine(layer []VisualNode, maxWidth int) string {
-	// Build a line of spaces
-	line := strings.Repeat(" ", maxWidth)
-	lineRunes := []rune(line)
+	if !f.colorConfig.ColorsEnabled {
+		// If colors are disabled, use the simple approach
+		line := strings.Repeat(" ", maxWidth)
+		lineRunes := []rune(line)
 
-	// Track which positions have been colored to avoid overlaps
-	coloredPositions := make(map[int]bool)
+		// Track which positions have been colored to avoid overlaps
+		coloredPositions := make(map[int]bool)
 
-	// Place each value with its color
-	for _, node := range layer {
-		value := node.Position.Value
-		startPos := node.PipePosition
+		// Place each value
+		for _, node := range layer {
+			value := node.Position.Value
+			startPos := node.PipePosition
 
-		// Place the raw value in the line first
-		if startPos < len(lineRunes) {
-			// Clear the space for this value
-			valueRunes := []rune(value)
-			for i, r := range valueRunes {
-				if startPos+i < len(lineRunes) && !coloredPositions[startPos+i] {
-					lineRunes[startPos+i] = r
-					coloredPositions[startPos+i] = true
+			// Place the raw value in the line
+			if startPos < len(lineRunes) {
+				valueRunes := []rune(value)
+				for i, r := range valueRunes {
+					if startPos+i < len(lineRunes) && !coloredPositions[startPos+i] {
+						lineRunes[startPos+i] = r
+						coloredPositions[startPos+i] = true
+					}
 				}
 			}
 		}
+
+		return string(lineRunes)
 	}
 
-	// Convert back to string and apply colors if enabled
-	plainLine := string(lineRunes)
+	// When colors are enabled, build the line more carefully
+	// to avoid ANSI escape sequence corruption
+	var result strings.Builder
+	result.Grow(maxWidth * 2) // Pre-allocate space for colors
 
-	if !f.colorConfig.ColorsEnabled {
-		return plainLine
-	}
+	// Sort nodes by position to place them left-to-right
+	sortedNodes := make([]VisualNode, len(layer))
+	copy(sortedNodes, layer)
+	sort.Slice(sortedNodes, func(i, j int) bool {
+		return sortedNodes[i].PipePosition < sortedNodes[j].PipePosition
+	})
 
-	// Apply colors to the line by replacing values with colored versions
-	result := plainLine
-	for _, node := range layer {
+	currentPos := 0
+	for _, node := range sortedNodes {
 		value := node.Position.Value
+		startPos := node.PipePosition
+
+		// Add spaces before this value
+		if startPos > currentPos {
+			spacesNeeded := startPos - currentPos
+			result.WriteString(strings.Repeat(" ", spacesNeeded))
+		}
+
+		// Add the colored value
 		isOperator := f.isOperatorValue(node.Position.Expression, value)
 		coloredValue := f.colorizeValue(value, isOperator)
+		result.WriteString(coloredValue)
 
-		// Replace the first occurrence of the value with its colored version
-		result = strings.Replace(result, value, coloredValue, 1)
+		// Update current position (visual position, not including ANSI sequences)
+		currentPos = startPos + len([]rune(value))
 	}
 
-	return result
+	// Add trailing spaces if needed
+	if currentPos < maxWidth {
+		result.WriteString(strings.Repeat(" ", maxWidth-currentPos))
+	}
+
+	return result.String()
 }
 
 // isOperatorValue determines if a value represents an operator result

--- a/internal/formatter/visual_test.go
+++ b/internal/formatter/visual_test.go
@@ -277,3 +277,73 @@ func TestVisualFormatter_NoMachineReadable(t *testing.T) {
 		t.Error("Machine-readable section should not be included when disabled")
 	}
 }
+
+func TestVisualFormatter_Issue10_Alignment(t *testing.T) {
+	// Test for GitHub issue #10 - alignment issues with colored output
+	formatter := NewVisualFormatter()
+
+	result := &evaluator.ExpressionResult{
+		Expression: "result == expected",
+		Result:     false,
+		Variables: map[string]interface{}{
+			"result":   3,
+			"expected": 0,
+		},
+		Tree: &evaluator.EvaluationTree{
+			ID:       1,
+			Type:     "comparison",
+			Operator: "==",
+			Text:     "result == expected",
+			Result:   false,
+			Left: &evaluator.EvaluationTree{
+				ID:    2,
+				Type:  "identifier",
+				Text:  "result",
+				Value: 3,
+			},
+			Right: &evaluator.EvaluationTree{
+				ID:    3,
+				Type:  "identifier",
+				Text:  "expected",
+				Value: 0,
+			},
+		},
+	}
+
+	output := formatter.FormatVisual(result, "test.go", 1, "")
+
+	// Verify the output doesn't contain truncated ANSI escape sequences
+	if strings.Contains(output, "30m") || strings.Contains(output, "31m") || strings.Contains(output, "32m") {
+		t.Errorf("Output contains truncated ANSI escape sequences: %q", output)
+	}
+
+	// Verify that values are properly aligned
+	lines := strings.Split(output, "\n")
+	var assertLine, valueLine string
+	for i, line := range lines {
+		if strings.Contains(line, "assert(result == expected)") {
+			assertLine = line
+			// Find a line with values
+			for j := i + 1; j < len(lines); j++ {
+				if strings.Contains(lines[j], "3") || strings.Contains(lines[j], "0") {
+					valueLine = lines[j]
+					break
+				}
+			}
+			break
+		}
+	}
+
+	if assertLine == "" {
+		t.Error("Could not find assert line in output")
+	}
+	if valueLine == "" {
+		t.Error("Could not find value line in output")
+	}
+
+	// The test mainly checks that we don't get truncated escape sequences
+	// and that the output is well-formed
+	t.Logf("Assert line: %q", assertLine)
+	t.Logf("Value line: %q", valueLine)
+	t.Logf("Full output:\n%s", output)
+}

